include/caffe/data_transformer.hpp

```diff
@@ -16,7 +16,7 @@ namespace caffe {
 template <typename Dtype>
 class DataTransformer {
  public:
-  explicit DataTransformer(const TransformationParameter& param);
+  explicit DataTransformer(const TransformationParameter& param, Phase phase);
```

Comment:
Maybe it would make sense to add `phase` to `TransformationParameter`, setting the phase from the owning net by default? (Keeps the single-argument constructor, and allows one to override defaults, e.g., if one wanted to randomly crop/mirror at test time for some reason.)